packages/validate/src/lib/validation.ts

```diff
@@ -141,9 +173,17 @@ export class Validator<T> implements Validatable<T> {
 	 * if the value can't be converted to the new type, or return the new type otherwise.
 	 */
 	refine<U>(otherValidationFn: (value: T) => U): Validator<U> {
-		return new Validator((value) => {
-			return otherValidationFn(this.validate(value))
-		})
+		// TODO: is this OK? that it doesn't use the knownGoodVersion function?
```

Comment:
this function originally had two purposes:
1. allow type refinements that don't change the value (e.g. `T.string.refine(validateShapeId)` becomes a `Validator<ShapeId>`)
2. allow different types at runtime than at rest (e.g. `T.object({x: ..., y: ...}).refine(Vec.From)` becomes a `Validator<Vec>`

2 went away early on in the process because we decided that validator function shouldn't operate as parsers in this way - I think there are checks that calls to `.validate` return identical values. So this should be fine!

Reply:
cool thanks. I will remove that TODO and make sure I test the (1) behavior.